### PR TITLE
Switch to server-side Apply on cluster create

### PR DESCRIPTION
**Pull request by csrwng**
Ensures that the content of existing secrets with the same name are overwritten with whatever is getting created.

### CODE REVIEW

File: cmd/cluster/create.go [hunk 1]
```diff
@@ -253,11 +253,11 @@ func CreateCluster(ctx context.Context, opts Options) error {
 	default:
 		for _, object := range exampleObjects {
 			key := crclient.ObjectKeyFromObject(object)
-			_, err = controllerutil.CreateOrUpdate(ctx, client, object, NoopReconcile)
```
Comment (enxebre) [inline cmd/cluster/create.go:256]: nit: we can inline this if now

**Comment by enxebre** [comment]
We'll want to use https://github.com/kubernetes-sigs/controller-runtime/pull/1474 for all our patching operations once it makes it through.

File: cmd/cluster/create.go (hunk 1 above)
Comment (csrwng) [reply cmd/cluster/create.go:256]: thx, fixed

**Comment by sjenning** [comment]
/lgtm
/hold
for e2e completion

**Comment by csrwng** [comment]
Previous test run failed with:
```
[36mINFO[0m[2021-04-21T14:39:01Z] time="2021-04-21T14:38:33Z" level=info msg="found stack event" LogicalResourceId=example-ptj2w-65qnv ResourceStatus=ROLLBACK_IN_PROGRESS ResourceStatusReason="The following resource(s) failed to create: [SubdomainPublicZone, OIDCImageRegistryRole, OIDCBucket, WorkerInstanceProfile, OIDCIngressRole, OIDCCSIDriverRole, NodePoolManagementUser, KubeCloudControllerUser]. Rollback requested by user." ResourceType="AWS::CloudFormation::Stack" 
```

**Pull request merged**